common/spec/dependabot/pull_request_creator/branch_namer_spec.rb

```diff
@@ -247,6 +247,43 @@
       end
     end
 
+    context "with a removed transitive dependency" do
+      let(:dependencies) { [removed_dep, parent_dep] }
+      let(:removed_dep) do
+        Dependabot::Dependency.new(
+          name: "business",
+          version: nil,
+          previous_version: "1.4.0",
+          package_manager: "dummy",
+          requirements: [],
+          previous_requirements: [],
+          removed: true
+        )
+      end
+      let(:parent_dep) do
+        Dependabot::Dependency.new(
+          name: "statesman",
+          version: "1.5.0",
+          previous_version: "1.4.0",
+          package_manager: "dummy",
+          requirements: [{
+            file: "Gemfile",
+            requirement: "~> 1.5.0",
+            groups: [],
+            source: nil
+          }],
+          previous_requirements: [{
+            file: "Gemfile",
+            requirement: "~> 1.4.0",
+            groups: [],
+            source: nil
+          }]
+        )
+      end
+
+      it { is_expected.to eq("dependabot/dummy/business-and-statesman-") }
```

Comment:
What do you think about `dependabot/dummy/business-and-statesman--removed`? I think using two dashes is a nice middleground that would signal the dependency as removed _and_ prevent conflicts.

Reply:
I like it as that makes a conflict much less likely. Updated in https://github.com/dependabot/dependabot-core/pull/5623/commits/e0dabbcab9a9f81a3a892b3185f96c230e39c015.